### PR TITLE
filter tests moved to pytest

### PR DESCRIPTION
**Pull request by pondrejk**
```
pytest tests/foreman/api/test_filter.py 
================================================= test session starts =================================================

tests/foreman/api/test_filter.py ...                                                                            [100%]

=========================================== 3 passed, 12 warnings in 13.71s ===========================================
```
```

pytest tests/foreman/cli/test_filter.py 
================================================= test session starts =================================================                                                                                                 

tests/foreman/cli/test_filter.py ........                                                                       [100%]

============================================ 8 passed in 290.12s (0:04:50) ============================================

```

### CODE REVIEW

File: tests/foreman/cli/test_filter.py [hunk 1]
```diff
@@ -23,196 +23,204 @@
...
-        )
-        filter_ = Filter.info({'id': filter_['id']})
-        # We expect here only one organization and location
-        self.assertEqual(filter_['organizations'][0], new_org['name'])
-        self.assertEqual(filter_['locations'][0], new_loc['name'])
+
+
+@pytest.fixture(scope='module')
+def module_perms():
+    """Search for provisioning template permissions. Set ``cls.ct_perms``."""
```
Comment (swadeley) [inline tests/foreman/cli/test_filter.py:30]:
```suggestion
    """Search for provisioning template permissions. Set ``ct_perms``."""
```

**Review by swadeley — APPROVED** [review]
ACK

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_filter.py

File: tests/foreman/cli/test_filter.py [hunk 2]
```diff
@@ -23,196 +23,204 @@
...
+        for permission in Filter.available_permissions({"search": "resource_type=User"})
+    ]
+    return perms
+
+
+@pytest.fixture(scope='function')
+def function_role():
+    """Create a role that a filter would be assigned """
+    role = make_role()
+    return role
```
Comment (mshriver) [inline tests/foreman/cli/test_filter.py:42]: Bit of code golf, but a good practice to follow - there's no need to assign `role` here, just ` return make_role()`

**Review by mshriver — CHANGES_REQUESTED** [review]
One small change concerning an assertion pattern of two lists, otherwise looks great.

**Review by mshriver — APPROVED** [review]

**Pull request merged**